modules/nextflow/src/main/groovy/nextflow/container/CharliecloudCache.groovy

```diff
@@ -79,11 +80,14 @@ class CharliecloudCache {
     String simpleName(String imageUrl) {
         def p = imageUrl.indexOf('://')
         def name = p != -1 ? imageUrl.substring(p+3) : imageUrl
-
-        // add registry
-        if( registry )
-            name = registry + name
         
+        // add registry
+        if( registry ) {
+            if( !registry.endsWith('/') ) {
+                registry += '/'
+            }
+            imageUrl = registry + imageUrl
```

Comment:
Is this correct? why updating `imageUrl` if it's not used

Reply:
Good catch, it is incorrect, should be `name`. I will update.